flood_adapt/object_model/tipping_point.py

```diff
@@ -0,0 +1,338 @@
+        tp_results = pd.read_csv(tp_path / "tipping_point_results.csv")
+
+        for metric in self.attrs.tipping_point_metric:
+            fig = go.Figure()
+
+            fig.add_trace(
+                go.Scatter(
+                    x=self.attrs.sealevelrise,
+                    y=tp_results[f"{metric[0]}_value"],
+                    mode="lines+markers",
+                    name=f"{metric[0]}",
+                )
+            )
+
+            fig.add_shape(
+                type="line",
+                x0=0,
+                x1=1,
+                y0=metric[1],
+                y1=metric[1],
+                xref="paper",
+                yref="y",
+                line=dict(
+                    color="Red",
+                    width=2,
+                    dash="dash",
+                ),
+                name=f"{metric[0]} threshold",
+            )
+            fig.update_layout(
+                title=f"Tipping Point Analysis for {self.attrs.name}",
+                xaxis_title="Sea Level Rise (m)",
+                yaxis_title="Value",
+            )
+
+            fig.show()
+
+    ### standard functions ###
+    def load_file(filepath: Union[str, Path]) -> "TippingPoint":
+        """Create risk event from toml file."""
+        obj = TippingPoint()
+        with open(filepath, mode="rb") as fp:
+            toml = tomli.load(fp)
+        obj.attrs = TippingPointModel.model_validate(toml)
+        return obj
+
+    def load_dict(dct: Union[str, Path]) -> "TippingPoint":
+        """Create risk event from toml file."""
+        obj = TippingPoint()
+        obj.attrs = TippingPointModel.model_validate(dct)
+        return obj
+
+    def save(self, filepath: Union[str, os.PathLike]):
+        """Save tipping point to a toml file."""
+        with open(filepath, "wb") as f:
+            tomli_w.dump(self.attrs.model_dump(exclude_none=True), f)
+
+    def __eq__(self, other):
+        if not isinstance(other, TippingPoint):
+            # don't attempt to compare against unrelated types
+            raise NotImplementedError
+        attrs_1, attrs_2 = self.attrs.model_copy(), other.attrs.model_copy()
+        attrs_1.__delattr__("name"), attrs_2.__delattr__("name")
+        attrs_1.__delattr__("description"), attrs_2.__delattr__("description")
+        return attrs_1 == attrs_2
+
+
+# TODO: post processing stuff still to be done for frontend
+# make html & plots
+# write to file
+
+if __name__ == "__main__":
+    from flood_adapt.config import set_system_folder
+
+    database = read_database(
+        r"C:\\Users\\morenodu\\OneDrive - Stichting Deltares\\Documents\\GitHub\\Database",
+        "charleston_test",
+    )
+    set_system_folder(
+        r"C:\\Users\\morenodu\\OneDrive - Stichting Deltares\\Documents\\GitHub\\Database\\system"
+    )
+
+    tp_dict = {
+        "name": "tipping_point_test",
+        "description": "",
+        "event_set": "extreme12ft",
+        "strategy": "no_measures",
```

Comment:
@LuukBlom, also something important to consider, but I still have to think better on what is the best solution: they use different strategies in the same place (as seen in the table), whereas we have a single strategy for all the scenarios. Should we adapt the tipping point object or make a wrapper to combine all these in plots and tables?